### ISSUE TITLE
Node hook install failures

### ISSUE THREAD

**Issue by acdha** [pre-commit/pre-commit#2433]
### search tried in the issue tracker

npm, node, "no matching version found"

### describe your issue

Something appears to have changed recently which breaks NPM-based hooks on my macOS system using the Homebrew pre-commit package. I've been using pre-commit for years but recently every repo I have has started failing the first time it installs a new version of a hook which uses Node (e.g. Prettier, ESLint, etc.).

They all fail with an error message like this referencing whatever version of that package was produced by `pre-commit autoupdate`:

```
An unexpected error has occurred: CalledProcessError: command: ('/Users/cadams/.cache/pre-commit/repo130gxljk/node_env-system/bin/node', '/opt/homebrew/bin/npm', 'install', '-g', '/Users/cadams/.cache/pre-commit/repo130gxljk/placeholder_package-0.0.0.tgz', 'prettier@2.7.1')
return code: 1
expected return code: 0
stdout: (none)
stderr:
    npm ERR! code ETARGET
    npm ERR! notarget No matching version found for prettier@2.7.1.
```

Of note is that the path in the first argument references a `node_env-system` directory hierarchy which does not exist.

```fish
total 32
-rw-r--r--  1 cadams  staff  1059 Jun 29 15:27 LICENSE
-rw-r--r--  1 cadams  staff  1049 Jun 29 15:27 README.md
-rw-r--r--  1 cadams  staff   133 Jun 29 15:27 package.json
-rw-r--r--  1 cadams  staff  1290 Jun 29 15:29 placeholder_package-0.0.0.tgz
```

```fish
cadams@Galatea ~> npm --version 
8.12.1
```

### pre-commit --version

pre-commit 2.19.0

### .pre-commit-config.yaml

```yaml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
      - id: check-added-large-files
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.7.1
    hooks:
      - id: prettier
        files: \.(css|less|scss|ts|tsx|graphql|gql|js|jsx|md)$
  - repo: https://github.com/pre-commit/mirrors-eslint
    rev: v8.18.0
    hooks:
      - id: eslint
```

### version information

```
pre-commit version: 2.19.0
git --version: git version 2.37.0
sys.version:
    3.10.5 (main, Jun 23 2022, 17:14:57) [Clang 13.1.6 (clang-1316.0.21.2.5)]
sys.executable: /opt/homebrew/Cellar/pre-commit/2.19.0/libexec/bin/python3.10
os.name: posix
sys.platform: darwin
```

### error information

```
An unexpected error has occurred: CalledProcessError: command: ('/Users/cadams/.cache/pre-commit/repo130gxljk/node_env-system/bin/node', '/opt/homebrew/bin/npm', 'install', '-g', '/Users/cadams/.cache/pre-commit/repo130gxljk/placeholder_package-0.0.0.tgz', 'prettier@2.7.1')
return code: 1
expected return code: 0
stdout: (none)
stderr:
    npm ERR! code ETARGET
    npm ERR! notarget No matching version found for prettier@2.7.1.
    npm ERR! notarget In most cases you or one of your dependencies are requesting
    npm ERR! notarget a package version that doesn't exist.
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /Users/cadams/.npm/_logs/2022-06-29T19_29_42_809Z-debug-0.log
    
```

```
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/error_handler.py", line 73, in error_handler
    yield
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/main.py", line 389, in main
    return run(args.config, store, args)
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/commands/run.py", line 414, in run
    install_hook_envs(to_install, store)
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/repository.py", line 223, in install_hook_envs
    _hook_install(hook)
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/repository.py", line 79, in _hook_install
    lang.install_environment(
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/languages/node.py", line 117, in install_environment
    helpers.run_setup_cmd(prefix, install)
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/languages/helpers.py", line 51, in run_setup_cmd
    cmd_output_b(*cmd, cwd=prefix.prefix_dir, **kwargs)
  File "/opt/homebrew/Cellar/pre-commit/2.19.0/libexec/lib/python3.10/site-packages/pre_commit/util.py", line 146, in cmd_output_b
    raise CalledProcessError(returncode, cmd, retcode, stdout_b, stderr_b)
pre_commit.util.CalledProcessError: command: ('/Users/cadams/.cache/pre-commit/repo130gxljk/node_env-system/bin/node', '/opt/homebrew/bin/npm', 'install', '-g', '/Users/cadams/.cache/pre-commit/repo130gxljk/placeholder_package-0.0.0.tgz', 'prettier@2.7.1')
return code: 1
expected return code: 0
stdout: (none)
stderr:
    npm ERR! code ETARGET
    npm ERR! notarget No matching version found for prettier@2.7.1.
    npm ERR! notarget In most cases you or one of your dependencies are requesting
    npm ERR! notarget a package version that doesn't exist.
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /Users/cadams/.npm/_logs/2022-06-29T19_29_42_809Z-debug-0.log
```

**Comment by acdha** [comment]
Ah, that explains the empty directory. I did have one `~/.npmrc` value set after running `npm config --global set prefer-offline true` during some network issues a few weeks back. I thought that was the cause because things worked after I removed that file entirely but when I restored it & removed `~/.cache/pre-commit` the test repo I was using yesterday still works after having failed consistently for the last week or so. 

Since this can't be reproduced any more I'll close the issue. I'm a bit concerned about not having a root cause, especially since I was using `npm` directly without noticing problems like this.
